dataplane/ebpf/src/main.rs

```diff
@@ -98,77 +60,36 @@ fn try_tc_ingress(ctx: TcContext) -> Result<i32, i64> {
         .load::<u8>(ETH_HDR_LEN + offset_of!(iphdr, protocol))
         .map_err(|_| TC_ACT_PIPE)?;
 
-    if protocol != IPPROTO_UDP {
-        return Ok(TC_ACT_PIPE);
-    }
-
-    let ip_hdr: *mut iphdr = unsafe { ptr_at(&ctx, ETH_HDR_LEN) }?;
-
-    let udp_header_offset = ETH_HDR_LEN + IP_HDR_LEN;
-
-    let udp_hdr: *mut udphdr = unsafe { ptr_at(&ctx, udp_header_offset)? };
-
-    let key = BackendKey {
-        ip: u32::from_be(unsafe { (*ip_hdr).daddr }),
-        port: (u16::from_be(unsafe { (*udp_hdr).dest })) as u32,
-    };
-
-    let backend = get_backend(key).ok_or(TC_ACT_OK)?;
-
-    let daddr_dot_dec = ip_from_int(unsafe { (*ip_hdr).daddr });
-    info!(
-        &ctx,
-        "Received a packet destined for svc ip: {}.{}.{}.{} at port: {}",
-        daddr_dot_dec[0],
-        daddr_dot_dec[1],
-        daddr_dot_dec[2],
-        daddr_dot_dec[3],
-        u16::from_be(unsafe { (*udp_hdr).dest })
-    );
-
-    // Update destination IP
-    unsafe {
-        (*ip_hdr).daddr = backend.daddr.to_be();
+    match protocol {
+        IPPROTO_TCP => handle_tcp_ingress(ctx),
```

Comment:
I think it's still smarter to handle any errors we may encounter in our `handle` functions here rather than internally

Reply:
(i.e like we still do for `tc_egress`)

Reply:
Oh nm we do :) in `tc_ingress` my bad